### ISSUE TITLE
stack path --ghc-paths broken

### ISSUE THREAD

**Issue by mboes** [commercialhaskell/stack#1546]
From the documentation:

```
  --ghc-paths              Installed GHCs (unpacked and archives)
```

I'm not sure how to grok the explanation of the option, but at any rate I'd expect this variable is essentially pointing me to the local GHC install that stack is using for the current project (or indeed the system installed GHC if that's what is being used).

In reality, in the `stack` repository I get:

```
/home/mboes/.stack/programs/x86_64-linux
```

which is a path that in fact does not exist (on my system).

**Comment by mgsloan** [comment]
`stack path --ghc-paths` tells you where `stack setup` will install GHC and other programs.  So, if `stack setup` has never been used, this path won't exist.  I agree that this is a confusing thing for `stack path --ghc-paths` to do.  I propose:

* Deprecating `--ghc-paths`, and adding `--programs-path` to provide the current functionality.
* Adding `--ghc-path`, which points to the GHC install used for the project.  Would you expect the value of `stack --ghc-path` to be something like `/usr/local/bin/ghc`, or just `/usr/local/`?

**Comment by sjakobi** [comment]
> Deprecating --ghc-paths, and adding --programs-path to provide the current functionality. Or maybe just straight up removing it.

I think that the flag loses most of it's value when there's also `--ghc-path`. Removing it should be ok.

> Would you expect the value of stack --ghc-path to be something like /usr/local/bin/ghc, or just /usr/local/?

Personally, I would prefer the more precise result.

What should `stack path --ghc-path` return in the case of a GHCJS project? If it should point at the location of `ghcjs` in that case, should the option possibly be called `--compiler-path` instead?

**Comment by mgsloan** [comment]
Good point!  I'd be in favor of having it be `--compiler-path`

**Comment by sjakobi** [comment]
Fixed via #2016.
